input: remove legacy form input export

JonasBa opened the pull request:
Undoes changes and fixes the root cause of required prop not being forwarded inside form components

JonasBa committed:
input: remove legacy form input export

JonasBa committed:
forms: manage required prop inside forms

JonasBa committed:
forms: fix import

JonasBa committed:
input: remove required

JonasBa committed:
input: deep in the weeds of form inputs

JonasBa committed:
input: add required back to deprecated form

JonasBa committed:
input: revert field prop

JonasBa committed:
input: fix input required forwarding (#87516)

This was never working because the input component was omitting the
required prop to the actual native element

scttcper reviewed (approved):
i don't have enough context on the legacy form export but this looks good

JonasBa commented:
@scttcper I'm going to check if we have other instances of Input component where we pass required that may not be tested, and remove it.

JonasBa committed:
input: required was a lie (#87543)

Unfortunately, the Input component has lied to us, and was
[never](https://github.com/getsentry/sentry/blob/a3096e5c1dafb4a2afec2fbe62a5dfae7a1881df/static/app/components/core/input/index.tsx#L91)
actually forwarding the required prop to the native input component...

Fixing this unfortunately creates a breaking change in the functionality
of some forms, some of which are untested. however given that this prop
had never worked, I have taken the liberty to remove it, and the authors
of the code that relied on it can add it back.
